x-pack/plugins/cases/server/client/mocks.ts

```diff
@@ -5,115 +5,81 @@
-
-  const caseConfigureService = await caseConfigureServicePlugin.setup();
-
-  const connectorMappingsService = await connectorMappingsServicePlugin.setup();
-  const userActionService = {
-    getUserActions: jest.fn(),
-    postUserActions: jest.fn(),
+};
+
+type AttachmentsSubClientMock = jest.Mocked<AttachmentsSubClient>;
+
+const createAttachmentsSubClientMock = (): AttachmentsSubClientMock => {
+  return {
+    add: jest.fn(),
   };
+};
 
-  const alertsService = {
-    initialize: jest.fn(),
-    updateAlertsStatus: jest.fn(),
-    getAlerts: jest.fn(),
+type UserActionsSubClientMock = jest.Mocked<UserActionsSubClient>;
+
+const createUserActionsSubClientMock = (): UserActionsSubClientMock => {
+  return {
+    getAll: jest.fn(),
   };
+};
 
-  // since the cases saved objects are hidden we need to use getScopedClient(), we'll just have it return the mock client
-  // that is passed in to createRouteContext
-  const savedObjectsService = savedObjectsServiceMock.createStartContract();
-  savedObjectsService.getScopedClient.mockReturnValue(savedObjectsClient);
-
-  // create a fake feature
-  const featureStart = featuresPluginMock.createStart();
-  featureStart.getKibanaFeatures.mockReturnValue([
-    // all the authorization class cares about is the `cases` field in the kibana feature so just cast it to that
-    ({ cases: ['securitySolution'] } as unknown) as KibanaFeature,
-  ]);
-
-  const factory = new CasesClientFactory(log);
-  factory.initialize({
-    alertsService,
-    caseConfigureService,
-    caseService,
-    connectorMappingsService,
-    userActionService,
-    featuresPluginStart: featureStart,
-    getSpace: async (req: KibanaRequest) => undefined,
-    // intentionally not passing the security plugin so that security will be disabled
-  });
-
-  // create a single reference to the caseClient so we can mock its methods
-  const casesClient = await factory.create({
-    savedObjectsService,
-    // Since authorization is disabled for these unit tests we don't need any information from the request object
-    // so just pass in an empty one
-    request: {} as KibanaRequest,
-    scopedClusterClient: esClient,
-  });
+type SubCasesClientMock = jest.Mocked<SubCasesClient>;
 
+const createSubCasesClientMock = (): SubCasesClientMock => {
   return {
-    client: casesClient,
-    services: { userActionService, alertsService },
-    esClient,
+    delete: jest.fn(),
+    find: jest.fn(),
+    get: jest.fn(),
+    update: jest.fn(),
+  };
+};
+
+type CasesClientInternalMock = jest.Mocked<CasesClientInternal>;
+
+export interface CasesClientMock extends CasesClient {
+  cases: CasesSubClientMock;
+  attachments: AttachmentsSubClientMock;
+  userActions: UserActionsSubClientMock;
+  subCases: SubCasesClientMock;
+}
+
+export const createCasesClientMock = (): CasesClientMock => {
+  const client: PublicContract<CasesClient> = {
+    casesClientInternal: (jest.fn() as unknown) as CasesClientInternalMock,
```

Comment:
When my PR gets merged we can remove this.

Reply:
And we can also add the `stats` subclient and the `getTags`, `getReporters` methods to the `CasesSubClientMock`